cnf-certification-test/accesscontrol/securitycontextcontainer/securitycontextcontainer.go

```diff
@@ -0,0 +1,343 @@
+		sort.Strings(sliceDropCapabilities)
+		logrus.Info("sliceDropCapabilities", sliceDropCapabilities)
+
+		sort.Strings(requiredDropCapabilities)
+		if reflect.DeepEqual(sliceDropCapabilities, requiredDropCapabilities) || reflect.DeepEqual(sliceDropCapabilities, dropAll) {
+			containerSCC.HaveDropCapabilities = true
+		}
+
+		if checkContainCategory(cut.SecurityContext.Capabilities.Add, category2AddCapabilities) {
+			logrus.Info("category is category2")
+			containerSCC.Capabilities = "category2"
+		} else {
+			if checkContainCategory(cut.SecurityContext.Capabilities.Add, category3AddCapabilities) {
+				containerSCC.Capabilities = "category3"
+			} else {
+				if len(cut.SecurityContext.Capabilities.Add) > 0 {
+					containerSCC.Capabilities = "category5"
+				} else {
+					logrus.Info("category is category1")
+					containerSCC.Capabilities = category1
+				}
+			}
+		}
+	} else {
+		containerSCC.Capabilities = category1
+	}
+	return containerSCC
+}
+
+func AllVolumeAllowed(volumes []corev1.Volume) bool {
+	countVolume := 0
+	for j := 0; j < len(volumes); j++ {
+		if volumes[j].ConfigMap != nil {
+			countVolume++
+		}
+		if volumes[j].DownwardAPI != nil {
+			countVolume++
+		}
+		if volumes[j].EmptyDir != nil {
+			countVolume++
+		}
+		if volumes[j].PersistentVolumeClaim != nil {
+			countVolume++
+		}
+		if volumes[j].Projected != nil {
+			countVolume++
+		}
+		if volumes[j].Secret != nil {
+			countVolume++
+		}
+	}
+	return countVolume == len(volumes)
+}
+
+type CategoryID int
+
+const (
+	Undefined CategoryID = iota
+	CategoryID1
+	CategoryID1NoUID0
+	CategoryID2
+	CategoryID3
+	CategoryID4
+)
+
+type PodListcategory struct {
+	Containername string
+	Podname       string
+	NameSpace     string
+	Category      CategoryID
+}
+
+const (
+	CategoryID1String       = "CategoryID1"
+	CategoryID1NoUID0String = "CategoryID1NoUID0"
+	CategoryID2String       = "CategoryID2"
+	CategoryID3String       = "CategoryID3"
+	CategoryID4String       = "OtherTypes"
+)
+
+// print the strings
+func (category CategoryID) String() string {
+	switch category {
+	case CategoryID1:
+		return CategoryID1String
+	case CategoryID1NoUID0:
+		return CategoryID1NoUID0String
+	case CategoryID2:
+		return CategoryID2String
+	case CategoryID3:
+		return CategoryID3String
+	case CategoryID4:
+		return CategoryID4String
+	case Undefined:
+		return CategoryID4String
+	}
+	return CategoryID4String
+}
+
+//nolint:funlen
+func CheckCategory(containers []corev1.Container, containerSCC ContainerSCC, podName, nameSpace string) []PodListcategory {
+	var ContainerList []PodListcategory
+	var categoryinfo PodListcategory
+	for j := 0; j < len(containers); j++ {
+		cut := &(containers[j])
+		percontainerSCC := GetContainerSCC(cut, containerSCC)
+		tnf.ClaimFilePrintf("percontainerSCC is ", percontainerSCC)
+		// after building the containerSCC need to check to which category it is
+		switch percontainerSCC {
+		case Category1:
```

Comment:
This is hard to debug without a log of why it was not equal. It would be nice to know what fields are mismatched

Reply:
i added that function that is printing to the claim file for each container the mismatched